File: python/strings/rot13.py
import sys

def rot13(text):
	shift = ""
	for _ in text:
		key = ord(_)
		if key >= ord('a') and key <= ord('m'):
			key += 13
		elif key >= ord('A') and key <= ord('M'):
			key += 13
		elif key >= ord('n') and key <= ord('z'):
			key -= 13
		elif key >= ord('n') and key <= ord('z'):
			key -= 13
		shift += chr(key)
	return shift
           
def main():
	if len(sys.argv) > 1:
		print(rot13(sys.argv[1]))
	else:
		x = str(input("Enter a string to encode or decode: "))
		print(rot13(x))


if __name__ == "__main__":
	main()

File: java/strings/ROT13.java
public class ROT13{
	public static void main(String[] args){
		String text = "This is some sample text";
		String rot13 = rot13(text);
		System.out.println("Before ROT13 shift: " + text);
		System.out.println("After ROT13 shift: " + reverse);
	}
	public static String rot13(String text){
		String shift = "";
		for (int i = 0; i < text.length(); i++) {
			char c = text.charAt(i);
			if       (c >= 'a' && c <= 'm') c += 13;
			else if  (c >= 'A' && c <= 'M') c += 13;
			else if  (c >= 'n' && c <= 'z') c -= 13;
			else if  (c >= 'N' && c <= 'Z') c -= 13;
			shift += c;
		}
		return shift;
	}
}